core/core.go

```diff
@@ -310,3 +311,35 @@ func (n *OpenBazaarNode) EncryptMessage(peerID peer.ID, peerKey *libp2p.PubKey,
 func (n *OpenBazaarNode) IPFSIdentityString() string {
 	return n.IpfsNode.Identity.Pretty()
 }
+
+// CheckErroredOrderMessages check if there are any order messages which can be retried
+func (n *OpenBazaarNode) CheckErroredOrderMessages(interval time.Duration) {
+	if interval == 0 {
+		return
+	}
+	ticker := time.NewTicker(interval)
+	go func() {
+		for range ticker.C {
+			msgs, err := n.Datastore.Messages().GetAllErrored()
+			if err != nil {
+				log.Error(err)
+			} else {
+				for _, m := range msgs {
+					if m.MsgErr.Error() == ErrInsufficientFunds.Error() {
+
+						// Get handler for this msg type
+						handler := n.Service.HandlerForMsgType(pb.Message_MessageType(m.MessageType))
+						if handler != nil {
+							// Dispatch handler
+							//rpmes, err := handler(m.PeerID, m.Message, nil)
+							//if err != nil {
+							//	log.Debugf("%s handle message error from %s: %s", pmes.MessageType.String(), mPeer.Pretty(), err)
+							//}
+						}
+
+					}
+				}
+			}
+		}
+	}()
```

Comment:
Please build this as a separate service which runs parallel to the node and manages it's own dependencies with a proper Start and Stop. See `RecordAgingNotifier` for an example of this pattern. Ensure that Stop handles the graceful shutdown of this goroutine and calls `ticker.Stop()` release that resource.